src/Model/Variant/Collection.php

```diff
@@ -255,7 +255,8 @@ protected function getChildCollectionMapAndList(): array {
         $childProductsList = [];
 
         $parentIds = array_map(function ($product) {
-            return $product->getId();
+            $linkField = $this->metadataPool->getMetadata(ProductInterface::class)->getLinkField();
```

Comment:
here and in other places: only load $linkField ones, there is no need to check it inside each array/foreach iteration, it will not change

Reply:
@aleksandrsm Updated the changes, please check